File: src/main/java/dev/vrba/slack/tenor/bot/service/slack/dto/SlackView.java
package dev.vrba.slack.tenor.bot.service.slack.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Getter
@Serdeable
@AllArgsConstructor
@RequiredArgsConstructor
public class SlackView {
    @Nullable
    @JsonProperty("id")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String id;

    @NonNull
    @JsonProperty("type")
    private final String type;

    @NonNull
    @JsonProperty("callback_id")
    private final String callback;

    @NonNull
    @JsonProperty("title")
    private final SlackTextBlock title;

    @NonNull
    @JsonProperty("blocks")
    private final List<SlackBlock> blocks;

    @Nullable
    @JsonProperty("private_metadata")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String metadata;

    @Nullable
    @JsonProperty("submit")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private SlackTextBlock submit;
}

File: src/main/kotlin/dev/vrba/slack/service/slack/dto/SlackView.kt
package dev.vrba.slack.service.slack.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class SlackView(
    @JsonProperty
    @JsonInclude(JsonInclude.Include.NON_NULL)
    val id: String? = null,
    @JsonProperty("type")
    val type: String,
    @JsonProperty("callback_id")
    val callback: String,
    @JsonProperty("title")
    val title: SlackTextBlock,
    @JsonProperty("blocks")
    val blocks: List<SlackBlock> = emptyList(),
    @JsonProperty("private_metadata")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    val metadata: String? = null,
    @JsonProperty("submit")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    val submit: SlackTextBlock? = null,
)
